src/libmongoc/src/mongoc/mongoc-scram.c

```diff
@@ -1027,7 +1029,7 @@ _mongoc_sasl_prep_impl (const char *name, const char *in_utf8, bson_error_t *err
    utf8_codepoints = bson_malloc (sizeof (uint32_t) * ((size_t) num_chars + 1u)); /* add one for trailing 0 value. */
    const char *c = in_utf8;
 
-   for (size_t i = 0; i < num_chars; ++i) {
+   mlib_foreach_urange (i, num_chars) {
```

Comment:
```suggestion
   mlib_foreach_irange (i, num_chars) {
```
With Clang 20.0.0:
```
warning: implicit conversion changes signedness: 'ssize_t' (aka 'long') to 'uintmax_t' (aka 'unsigned long') [-Wsign-conversion]
```
Similarly for other calls to `mlib_foreach_irange` in this scope which use `num_chars` (which is `ssize_t`).